src/Microsoft.Dnx.ApplicationHost/Program.cs

```diff
@@ -229,19 +229,26 @@ private Task<int> ExecuteMain(DefaultHost host, string applicationName, string[]
             {
                 assembly = host.GetEntryPoint(applicationName);
             }
-            catch (FileNotFoundException ex) when (new AssemblyName(ex.FileName).Name == applicationName)
+            catch (Exception ex)
             {
-                if (ex.InnerException is ICompilationException)
+                // Compilation exceptions either throw FileNotFound or FileLoad exceptions and may change
+                // from version to version. It's safest to catch both types of exceptions
+                if (ex is FileLoadException || ex is FileNotFoundException)
```

Comment:
We could, but historically I've seen us only catching these two types of exceptions. e.g. 196e111cc8e42cd79f616a962b86425b558512a6